### PR TITLE
Add new Twig namespace PrestaShopCore

### PR DESCRIPTION
**Pull request by atomiix**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Since the path syntax in twig using colon (`PrestaShopBundle:path:to:twig:file`) is not available anymore, the only way for a module to override **and** extends a core template is by using the `!` after the `@` in the namespace (`@!PrestaShop/path/to/twig/file`). Although this is working for 1.7.x & 8.x, this PR aims to add a new namespace (`@PrestaShopCore`) that modules **can't** override allowing modules to be sure to extends from the core file and not an override without the need to use the `!` (`@PrestaShopCore/path/to/twig/file`).
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| How to test?      | Install this module: [prestashopcore.zip](https://github.com/PrestaShop/PrestaShop/files/7753254/prestashopcore.zip) and refresh the module manager page. You should see Capture d’écran 2021-12-21 à 12 04 56
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27044)

### CODE REVIEW

File: app/config/config.yml [hunk 1]
```diff
@@ -87,6 +87,7 @@ twig:
     '%admin_page%/Configure/ShopParameters': ShopParameters
     '%kernel.root_dir%/../modules': Modules
     '%mail_themes_dir%': MailThemes
+    '%admin_page%/../': PrestaShopCore
```
Comment (jolelievre) [inline app/config/config.yml]: I would suggest adding a new parameter instead in config.yml
```yml
  admin_page: "%kernel.root_dir%/../src/PrestaShopBundle/Resources/views/Admin"
  prestashop_views: "%kernel.project_dir%/src/PrestaShopBundle/Resources/views"
```

Comment (jolelievre) [reply app/config/config.yml]:
```suggestion
    '%prestashop_views%': PrestaShopCore
```

Comment (jolelievre) [reply app/config/config.yml]: This will be useful when we start migrating the front to Symfony

**Review by jolelievre — DISMISSED** [review]
Thanks @atomiix 

The change is legitimate, I just suggested an alternative Feel free to use it or not 😉

**Review by jolelievre — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by PierreRambaud — COMMENTED** [review]
Something I don't understand is why you don't had to change the `prestashop.twig.modules.loader` service? :thinking:

**Comment by atomiix** [comment]
@PierreRambaud that's the trick! If we add the new namespace to `prestashop.twig.modules.loader`, twig will also look inside module's folders to find templates when using `@PrestaShopCore`.

**Comment by PierreRambaud** [comment]
> @PierreRambaud that's the trick! If we add the new namespace to `prestashop.twig.modules.loader`, twig will also look inside module's folders to find templates when using `@PrestaShopCore`.

What's the aim to add templates we can't override? Why we should do that?

**Comment by jolelievre** [comment]
@PierreRambaud the initial issue comes from a module which wants to override a core template
- the module includes the file which matches a naming convention
- this module's file is used instead of the core one
- if the module's file wants to extend the core template (because it only wants to override or extend one block in particular) then we end up in an infinite loop

To avoid this, you can use the `@!` syntax so that the override template can reference to its "parent" template (the one it's overriding). The purpose of this namespace is to offer similar use but via a dedicated namespace. This way instead of having a module's template that almost extends itself but use `!` to target the parent, you can use this new core namespace which will ALWAYS target the initial core template

**Comment by jolelievre** [comment]
TLDR; as @atomiix explained in the description:
`@PrestaShopCore/path/to/twig/file` is synonym of `@!PrestaShop/path/to/twig/file`

But since not everyone knows about the `@!` notation we thought it was nice to have a dedicated namespace to handle this.

**Comment by PierreRambaud** [comment]
@jolelievre Thanks for explanation, it's quite better now :P

**Review by PierreRambaud — APPROVED** [review]

**Comment by jolelievre** [comment]
I guess this one should be QA by dev right?

**Comment by atomiix** [comment]
Yeah, probably, but I added a module to simplify the test so I guess the QA team can also do it 🤷‍♂️

**Comment by jolelievre** [comment]
@atomiix you're amazing 😍

**Comment by PierreRambaud** [comment]
@jolelievre @atomiix it can be done by a dev as there is nothing functional except for dev stuff ^^

**Review by matthieu-rolland — APPROVED** [review]
I tested with and without your PR, it's working 👍 

Capture d’écran 2021-12-27 à 17 02 59

**Comment by matthieu-rolland** [comment]
thank you @atomiix  !
